src/javascripts/Services/referencedByViewColumnsService.js

```diff
@@ -0,0 +1,114 @@
+ngapp.service('referencedByViewColumnsService', function(settingsService) {
```

Comment:
definitely a lot of repetition here with the `columnsService`.  Perhaps we can figure out a way to extend the `columnsService` to store columns generically?  We can move the column prototypes to a `columnsFactory` and then adjust the functions to work with any view.